src/draw.js

```diff
@@ -91,47 +108,136 @@ Draw.prototype = extend(Control, {
         if (!this.lineStringCtrl.classList.contains('active')) {
           this.lineStringCtrl.dispatchEvent(event);
         }
-      break;
+        break;
       case 77: // (m) marker
         if (!this.markerCtrl.classList.contains('active')) {
           this.markerCtrl.dispatchEvent(event);
         }
-      break;
+        break;
       case 80: // (p) polygon
         if (!this.polygonCtrl.classList.contains('active')) {
           this.polygonCtrl.dispatchEvent(event);
         }
-      break;
+        break;
       case 83: // (s) square
         if (!this.squareCtrl.classList.contains('active')) {
           this.squareCtrl.dispatchEvent(event);
         }
-      break;
+        break;
+      case 27: // (escape) exit edit mode
+        if (this.editId) {
+          this._exitEdit();
+        }
+        break;
     }
   },
 
-  _onMouseDown(e) {
-    if (e.altKey) {
-      e.stopPropagation();
-      // TODO https://github.com/mapbox/mapbox-gl-js/issues/1264
-      // this._captureFeatures();
+  _onMouseDown(/*e*/) {
+    //if (e.altKey) {
+    //  e.stopPropagation();
+    //    TODO https://github.com/mapbox/mapbox-gl-js/issues/1264
+    //    this._captureFeatures();
+    //}
+  },
+
+  _onClick(e) {
+    var coords = DOM.mousePos(e, this._map._container);
+    this._map.featuresAt([coords.x, coords.y], { radius: 10 }, (err, features) => {
+      if (err) throw err;
+      else if (!features.length && this.editId) return this._exitEdit();
+      else if (!features.length) return;
+
+      var feature = features[0];
+      if (this.editId && this.editId !== feature.properties._drawid) this._exitEdit();
+      else if (this.editId === feature.properties._drawid) return;
+
+      this.editId = feature.properties._drawid;
+      coords = feature.geometry.coordinates;
+      var type = feature.geometry.type;
+
+      if (type === 'Point')
```

Comment:
Could we remove this mismatch between featureType and GeoJSON features and just have them be equal?